components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -4689,4 +4696,17 @@ public static boolean hasIntersection(String[] arr1, String[] arr2) {
 
         return false;
     }
+
+    private static boolean isLightweightAPIKeyGenerationEnabled() {
+        try {
+            APIManagerConfiguration config = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                    .getAPIManagerConfiguration();
+            String lightweightAPIKeyGenerationEnabled = config.getFirstProperty(APIConstants.LIGHTWEIGHT_API_KEY_GENERATION_ENABLED);
+            return Boolean.parseBoolean(lightweightAPIKeyGenerationEnabled);
```

Comment:
When the config element is missing from the file, the default value that we return is changed to false, as the  Boolean.parseBoolean(null) returns falls. 
```suggestion
if (StringUtils.isNotEmpty(lightweightAPIKeyGenerationEnabled)
            return Boolean.parseBoolean(lightweightAPIKeyGenerationEnabled);
```

Reply:
When the config element is missing from the file, It will assign the default value from the default.json